### DISCUSSION TITLE
Autoplay state to control the styles of the <play> button

### DISCUSSION THREAD

**Discussion by kdzone** [davidjerleke/embla-carousel#308] [Ideas] [answered]
Playback can be stopped via stopOnMouseEnter and other internal events.
How to change the styles of my `play` button?

I suggest making a state for autoplay and an event to change it.

**Comment by davidjerleke** [comment]
Hi @kdzone,

Thank you for your suggestion. If I’m understanding you correctly, you’re suggesting something along these lines?

```ts
autoplay.on('play', yourPlayCallback)

autoplay.on('stop', yourStopCallback)
```

Or did you have something different in mind?

Best,
David

**Reply by dweidner** [reply]
Hi @davidjerleke, would you accept a pull request for this feature? I could use these events to update the state of a pause button in a component I am working on.

**Comment by kdzone** [comment]
Yes - this is a great solution. 
It would be nice to add a function (property) active or playing in `Autoplay` (it doesn't cost anything).

**Reply by davidjerleke** [reply]
Thanks for the suggestion @kdzone. That makes sense. I don’t know when I will get an opportunity to work on this but I think it’s a good idea, and it should definitely be added to the plugin. 

Best,
David

**Comment by davidjerleke** [comment]
- @kdzone @dweidner this is now tracked in issue #619. Closing this discussion.

**Reply by davidjerleke** [reply, marked as answer]
@kdzone @dweidner this feature has been included in the [example](https://www.embla-carousel.com/examples/predefined/#autoplay) in the docs now.
